server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1702,7 +1701,12 @@ public static PointInTimeBuilder fromXContent(XContentParser parser) throws IOEx
             if (params.id == null) {
                 throw new IllegalArgumentException("point int time id is not provided");
             }
-            return new PointInTimeBuilder(params.id, params.keepAlive);
+            return new PointInTimeBuilder(params.id).setKeepAlive(params.keepAlive);
+        }
+
+        public PointInTimeBuilder setKeepAlive(TimeValue keepAlive) {
```

Comment:
Can you add javadoc

Reply:
Added in https://github.com/elastic/elasticsearch/pull/62720/commits/5af72036b99e9f0c78f116948404f67af036fbe9